components/raftstore-v2/src/raft/peer.rs

```diff
@@ -366,14 +370,17 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
     /// Returns if there's any tombstone being removed.
     #[inline]
     pub fn remove_tombstone_tablets_before(&mut self, persisted: u64) -> bool {
-        let mut removed = 0;
-        while let Some(i) = self.pending_tombstone_tablets.first()
-            && *i <= persisted
-        {
-            removed += 1;
+        let removed = self
```

Comment:
This doesn't consider it's a monotonic sequence. If you must `drain_filter(..).next().is_some()` is better choice.

Reply:
The correctness of `take_while` relies on exactly monotonic.`drain_filter` will loop over all the numbers.